core/services/relay/evm/mercury/transmitter.go

```diff
@@ -0,0 +1,143 @@
+}
+
+type MercuryReport struct {
+	Payload     hexutil.Bytes
+	FromAccount common.Address
+}
+
+// Transmit sends the report to the on-chain smart contract's Transmit method.
+func (mt *MercuryTransmitter) Transmit(ctx context.Context, reportCtx ocrtypes.ReportContext, report ocrtypes.Report, signatures []ocrtypes.AttributedOnchainSignature) error {
+	var rs [][32]byte
+	var ss [][32]byte
+	var vs [32]byte
+	for i, as := range signatures {
+		r, s, v, err := evmutil.SplitSignature(as.Signature)
+		if err != nil {
+			panic("eventTransmit(ev): error in SplitSignature")
+		}
+		rs = append(rs, r)
+		ss = append(ss, s)
+		vs[i] = v
+	}
+	rawReportCtx := evmutil.RawReportContext(reportCtx)
+
+	payload, err := payloadTypes.Pack(rawReportCtx, []byte(report), rs, ss, vs)
+	if err != nil {
+		return errors.Wrap(err, "abi.Pack failed")
+	}
+
+	mr := MercuryReport{
+		Payload:     payload,
+		FromAccount: mt.fromAccount,
+	}
+	mt.lggr.Infow("Transmitting report", "mercuryReport", mr, "report", report, "reportCtx", reportCtx, "signatures", signatures)
+
+	b, err := json.Marshal(mr)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal mercury report JSON")
+	}
+
+	req, err := http.NewRequest(http.MethodPost, mt.reportURL, bytes.NewReader(b))
+	if err != nil {
+		return errors.Wrap(err, "failed to instantiate mercury server http request")
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.SetBasicAuth(mt.username, mt.password)
+	req = req.WithContext(ctx)
+
+	res, err := mt.httpClient.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "failed to POST to mercury server")
+	}
+	defer res.Body.Close()
+
+	// It's only used for logging, keep it short
+	safeLimitResp := http.MaxBytesReader(nil, res.Body, 1024)
+	respBody, err := io.ReadAll(safeLimitResp)
+	if err != nil {
+		mt.lggr.Errorw("Failed to read response body", "err", err)
+	}
+
+	if res.StatusCode >= 200 && res.StatusCode < 300 {
+		mt.lggr.Infow("Transmit report success", "responseStatus", res.Status, "reponseBody", string(respBody), "reportCtx", reportCtx)
+	} else {
+		mt.lggr.Errorw("Transmit report failed", "responseStatus", res.Status, "reponseBody", string(respBody), "reportCtx", reportCtx)
+
+	}
+
+	return nil
+}
+
+func (mt *MercuryTransmitter) FromAccount() ocrtypes.Account {
+	return ocrtypes.Account(mt.fromAccount.Hex())
+}
+
+// LatestConfigDigestAndEpoch retrieves the latest config digest and epoch from the OCR2 contract.
+// It is plugin independent, in particular avoids use of the plugin specific generated evm wrappers
+// by using the evm client Call directly for functions/events that are part of OCR2Abstract.
+func (mt *MercuryTransmitter) LatestConfigDigestAndEpoch(ctx context.Context) (cd ocrtypes.ConfigDigest, epoch uint32, err error) {
+	// ConfigDigest and epoch are not stored on the contract in mercury mode
+	// TODO: Do we need to support retrieving it from the server? Does it matter?
```

Comment:
I think these will matter, and will cause your ocr instance to fail.
You are still sharing the ConfigDigester, ContractTracker with regular EVM.
These will try to actually get config from EVM onchain, and digester will create digest from the contract config. That digest won't match with what is being returned from here.

Actually, you are returning an error from here, which may cause problems in the OCR protocol.
Have you confirmed with @secure-lorenz if such a hack can work?

Reply:
Yes, it works.